vivarium/core/engine.py

```diff
@@ -1013,6 +1022,9 @@ def run_for(
                         future = min(process_time + process_timestep, end_time)
                     else:
                         future = process_time + process_timestep
+                    if global_time_precision:
```

Comment:
This condition will evaluate to `False` when `global_time_precision` is zero, which someone might actually want (i.e. if they want to round the time to an integer)